### ISSUE TITLE
Add gitignore to all examples and generated from cli

### ISSUE THREAD

**Issue by masterbater** [honojs/hono#1364]
### What is the feature you are proposing?

Its better if you provide a default gitignore files, sometimes I forgot I didnt gitignore node_modules and some env files.

**Comment by yusukebe** [comment]
Hi @masterbater !

Did you say about the "[create hono](https://github.com/honojs/create-hono)" command?

**Comment by masterbater** [comment]
> Hi @masterbater !
> 
> Did you say about the "[create hono](https://github.com/honojs/create-hono)" command?

Yes, I use pnpm and it doesnt have any gitignore file in it

**Comment by masterbater** [comment]
> Hi @masterbater !
> 
> Did you say about the "[create hono](https://github.com/honojs/create-hono)" command?

Bun or nodejs has node_modules, I know its my responsibility to add gitignore, But I just make a quick spinning up a hono server and I notice when I git push it takes a while. 
<img width="659" alt="image" src="https://github.com/honojs/hono/assets/26339988/63ac30ec-0d94-4122-ba77-2757cd12c7c4">

**Comment by yusukebe** [comment]
@masterbater 

I agree with your observation. We should indeed include a `.gitignore` file. I've already set up the `.gitignore` for Cloudflare Workers and Pages templates. We'll need to integrate it into other projects, such as Bun, Node.js, and others.

Please feel free to open a PR addressing this in the "Create Hono" repository: https://github.com/honojs/starter. If not, I'll consider doing so.
